UI/Selenium/Utilities/DriverFactory.cs

```diff
@@ -72,36 +73,101 @@ public IWebDriver InitializeDriver(BrowserType browser)
+                        ,{"idleTimeout",sauceLabsOptions.IdleTimeoutInSeconds }
+                        ,{"maxDuration",sauceLabsOptions.MaxDurationInSeconds }
+                        ,{"seleniumVersion",sauceLabsOptions.SeleniumVersion }
+                        ,{"timeZone",sauceLabsOptions.Timezone }
+                    };
+            var remoteUrl = new Uri($"http://{config.SauceUsername}:{config.SauceAccessKey}{config.SauceUrl}");
 
-            foreach (var (key, value) in SauceOptions)
+            switch (config.PlatformName)
             {
-                options.AddAdditionalCapability(key, value);
-            }
+                case "Android":
+                    AppiumOptions options = new AppiumOptions();
+                    options.DeviceName = config.DeviceName;
+                    options.PlatformName = config.PlatformName;
+                    options.BrowserName = config.BrowserName;
+                    options.AddAdditionalAppiumOption(MobileCapabilityType.AppiumVersion, config.AppiumVersion);
+                    options.AddAdditionalAppiumOption(MobileCapabilityType.Orientation, config.Orientation);
+                    options.AddAdditionalAppiumOption("PlatformVersion", config.PlatformVersion);
+                    options.AddAdditionalAppiumOption("name", sauceLabsOptions.Name);
 
-            var remoteUrl = new Uri($"http://{config.SauceUsername}:{config.SauceAccessKey}{config.SauceUrl}");
-   
-            WebDriver = new RemoteWebDriver(remoteUrl, options.ToCapabilities());
+                    foreach (var (key, value) in SauceOptions)
+                    {
+                        options.AddAdditionalCapability(key, value);
+                    }
+                    WebDriver = new RemoteWebDriver(remoteUrl, options.ToCapabilities());
+                    break;
 
+                case "iOS":
+                    AppiumOptions iosOptions = new AppiumOptions();
+                    iosOptions.DeviceName = config.DeviceName;
+                    iosOptions.PlatformName = config.PlatformName;
+                    iosOptions.BrowserName = config.BrowserName;
+                    iosOptions.AddAdditionalAppiumOption(MobileCapabilityType.AppiumVersion, config.AppiumVersion);
+                    iosOptions.AddAdditionalAppiumOption(MobileCapabilityType.Orientation, config.Orientation);
+                    iosOptions.AddAdditionalAppiumOption("PlatformVersion", config.PlatformVersion);
+                    iosOptions.AddAdditionalAppiumOption("name", sauceLabsOptions.Name);
+
+                    foreach (var (key, value) in SauceOptions)
+                    {
+                        iosOptions.AddAdditionalCapability(key, value);
+                    }
+                    WebDriver = new RemoteWebDriver(remoteUrl, iosOptions.ToCapabilities());
+                    break;
+
+                case "macOS 12":
+                    if (config.BrowserName.Equals("chrome"))
```

Comment:
I see there is lot of if..else.. else ifs. The code can be simplified  by using
DriverOptions options